api/v1alpha1/frontend_types.go

```diff
@@ -58,6 +58,24 @@ type ServiceTile struct {
 	IsExternal bool   `json:"isExternal,omitempty" yaml:"isExternal,omitempty"`
 }
 
+type WidgetHeaderLink struct {
+	Title string `json:"title" yaml:"title"`
+	Href  string `json:"href" yaml:"href"`
+}
+
+type WidgetConfig struct {
+	Icon        string           `json:"icon" yaml:"icon"`
+	Title       string           `json:"title" yaml:"title"`
+	Permissions []Permission     `json:"permissions,omitempty" yaml:"permissions,omitempty"`
+	HeaderLink  WidgetHeaderLink `json:"headerLink,omitempty" yaml:"headerLink,omitempty"`
+}
+
+type WidgetEntry struct {
```

Comment:
@Hyperkid123 do you think we need to include the `Default`s for the widgets here too? See https://github.com/RedHatInsights/chrome-service-backend/blob/main/rest/service/dashboardTemplate.go#L29

Reply:
I think so and for each layout variant. Good catch.

Reply:
Agree. For now added the defaults to the struct